### DISCUSSION TITLE
mcuboot: how to integrate mcumgr into mcuboot ( as only the mcuboot is available  and mcumgr will write application)

### DISCUSSION THREAD

**Discussion by DineshVaghela** [zephyrproject-rtos/zephyr#81331] [Q&A] [answered]
I have a below scenario does mcuboot will help ?

I have only mcuboot running into the microcontroller. 
I need to load application into the external flash XIP with QSPI ( prior no application is available in slot 0 & 1)
can we integrate mcumgr in mcuboot code?
so that mcumgr takes app over serial and write to external flash with QSPI by mcuboot

**Comment by JPHutchins** [comment, marked as answer]
Yes, this is a normal use case, called "serial recovery" - though in some cases it is the ONLY mechanism to change FW.  See the Serial recovery section here: https://docs.mcuboot.com/readme-zephyr.html

Here are some sample configuration files that we use for HW integration tests.

* https://github.com/intercreate/smpclient/blob/main/dutfirmware/mcuboot_usb.conf
* https://github.com/intercreate/smpclient/blob/main/dutfirmware/mcuboot_usb.overlay

Since you have dual application slots, I do wonder if you really want the SMP server for firmware updates in the bootloader only or ALSO in the application?

**Reply by DineshVaghela** [reply]
Thanks for the reply I will try it out.
and in the Application , the firmware file will be parked in emmc and mcuboot will read and update it.so not require smp in application
